subsys/bluetooth/services/cgms/cgms_socp.c

```diff
@@ -0,0 +1,140 @@
+	SOCP_READ_PATIENT_LOW_ALERT_LEVEL_RESPONSE = 0x0C,
+	SOCP_SET_HYPO_ALERT_LEVEL = 0x0D,
+	SOCP_GET_HYPO_ALERT_LEVEL = 0x0E,
+	SOCP_HYPO_ALERT_LEVEL_RESPONSE = 0x0F,
+	SOCP_SET_HYPER_ALERT_LEVEL = 0x10,
+	SOCP_GET_HYPER_ALERT_LEVEL = 0x11,
+	SOCP_HYPER_ALERT_LEVEL_RESPONSE = 0x12,
+	SOCP_SET_RATE_OF_DECREASE_ALERT_LEVEL = 0x13,
+	SOCP_GET_RATE_OF_DECREASE_ALERT_LEVEL = 0x14,
+	SOCP_RATE_OF_DECREASE_ALERT_LEVEL_RESPONSE = 0x15,
+	SOCP_SET_RATE_OF_INCREASE_ALERT_LEVEL = 0x16,
+	SOCP_GET_RATE_OF_INCREASE_ALERT_LEVEL = 0x17,
+	SOCP_RATE_OF_INCREASE_ALERT_LEVEL_RESPONSE = 0x18,
+	SOCP_RESET_DEVICE_SPECIFIC_ALERT = 0x19,
+	SOCP_START_THE_SESSION = 0x1A,
+	SOCP_STOP_THE_SESSION = 0x1B,
+	SOCP_RESPONSE_CODE = 0x1C,
+};
+
+/**@brief Specific Operation Control Point response codes. */
+enum cgms_socp_rsp_code {
+	SOCP_RSP_RESERVED_FOR_FUTURE_USE = 0x00,
+	SOCP_RSP_SUCCESS = 0x01,
+	SOCP_RSP_OP_CODE_NOT_SUPPORTED = 0x02,
+	SOCP_RSP_INVALID_OPERAND = 0x03,
+	SOCP_RSP_PROCEDURE_NOT_COMPLETED = 0x04,
+	SOCP_RSP_OUT_OF_RANGE = 0x05,
+};
+
+static int generic_handler(struct bt_conn *peer, struct cgms_server *instance,
+		uint8_t req_opcode, uint8_t response_code)
+{
+	NET_BUF_SIMPLE_DEFINE(rsp, CGMS_SOCP_LENGTH);
+
+	net_buf_simple_add_u8(&rsp, SOCP_RESPONSE_CODE);
+	net_buf_simple_add_u8(&rsp, req_opcode);
+	net_buf_simple_add_u8(&rsp, response_code);
+
+	return cgms_socp_send_response(peer, &rsp);
+}
+
+static int write_cgms_communication_interval_handler(struct bt_conn *peer,
+		struct cgms_server *instance, const uint8_t *req_operand, uint16_t len)
+{
+	int rc;
+
+	if (len < 1) {
+		rc = generic_handler(peer, instance, SOCP_WRITE_CGM_COMMUNICATION_INTERVAL,
+					SOCP_RSP_INVALID_OPERAND);
+		return rc;
+	}
+
+	if (req_operand[0] > 0 && req_operand[0] < instance->comm_interval_minimum) {
+		rc = generic_handler(peer, instance, SOCP_WRITE_CGM_COMMUNICATION_INTERVAL,
+					SOCP_RSP_OUT_OF_RANGE);
+		return rc;
+	}
+
+	if (req_operand[0] == 0xFF) {
+		instance->comm_interval = instance->comm_interval_minimum;
+	} else {
+		/* In case of comm_interval = 0, we just change the value to 0,
+		 * and the peridoic notification task is responsible to examine
+		 * and apply the value correctly.
+		 */
+		instance->comm_interval = req_operand[0];
+	}
+
+	rc = generic_handler(peer, instance, SOCP_WRITE_CGM_COMMUNICATION_INTERVAL,
+					SOCP_RSP_SUCCESS);
+	return rc;
+}
+
+static int read_cgms_communication_interval_handler(struct bt_conn *peer,
+		struct cgms_server *instance)
+{
+	NET_BUF_SIMPLE_DEFINE(rsp, CGMS_SOCP_LENGTH);
+
+	net_buf_simple_add_u8(&rsp, SOCP_READ_CGM_COMMUNICATION_INTERVAL_RESPONSE);
+	net_buf_simple_add_le16(&rsp, instance->comm_interval);
+
+	return cgms_socp_send_response(peer, &rsp);
+}
+
+int cgms_socp_recv_request(struct bt_conn *peer, struct cgms_server *instance,
+			const uint8_t *req_data, uint16_t req_len)
+{
+	int rc;
+
+	/* Early check of the data length. */
+	if (req_data == NULL || req_len == 0) {
+		return -ENODATA;
+	}
+
```

Comment:
Same here:

Consider using net_buf_simple here. You can pull_u8 here for analyzing the first byte and pass the remaining content to downstream functions